### DISCUSSION TITLE
Copy tabs

### DISCUSSION THREAD

**Discussion by Cal63** [piroor/treestyletab#2824] [Ideas]
It would be a nice feature if you could right click on a tree and copy all it's tabs addresses to the clipboard.

**Comment by piroor** [comment]
A combination with [Copy Selected Tabs to Clipboard](https://addons.mozilla.org/firefox/addon/copy-selected-tabs-to-clipboar/) can be a workaround. Ctrl/Shift-click to select tabs of a tree and choose the copy format from the context menu, then you can copy tabs with arbitrary format. It supports indented output also.

**Reply by Casteele72** [reply]
`It` might be my Firefox, but this messes me up -- terribly. It does not let me 'select' tabs with precision, it only selects some tabs... and I cannot figure out the logic used. Sometimes it selects the tab alone, sometimes the tree starting with that tab, and sometimes only the partial tree. If I tree to drag to select more, it seems to randomly select some tabs dragged over, and skip others. And lastly, if I do manage to get the tabs I want selectedm I never get the option to copy them to clipboard: As soon as I try anything, it just makes a bunch of new tab copies of the select tabs in to TST, trashing my tree layout and order.

I would like to right-click a tab, select "Copy Tab Tree to Clipboard", and be done with it. Ideally, it would give me XML with the tree structure intact. Most similar tools use JSON, but JSON has some limits such as inability to have adjacent items with the same name (JavaScript was designed as a program scripting language, not a data description language. It simply assumes the 'identifier' is a redefinition of the previously defined one...); lack of ability to have comments (inline or EOL style); identifier 'names' which conform to JS rules but almost nothing else ("{['/*':true;'*/':false;]}" is valid JSON, but causes problems with many other tools and such...); among other issues I often have. (XML, OTOH, is robust and mature, and `has` many tools to manage XML, including converting to JSON -- where possible ((some XML things do not convert well to JSON, and v.v.))); And so on.

So far, the closest thing I have found to do what I want are some of the 'session manager' type extensions, but they all have similar problems (especially their use of JSON as an import/export data...).

My ideal would be to be able to copy the entire tree, including URI, page title (if any), and tree structure (parent-child relationships). Like (I hope this formats well, and using leading '.'s as indentation indicators to be easier to read):

```
<root>
....<tab>
........<uri>about:blank</uri>
........<title>about:blank</title>
........<children>
............<tab><uri>about:blank</uri><title>about:blank</title><children/></tab>
........</children>
....</tab>
</root>

```
(Of course, being XML, it does not need to be neatly visually formed like that, I did that just to help make my needs/thoughts visually clear.)

For years, I have been using 'Tab Session Manager', if you want to see how it keeps the data. It keeps WindowID and TabID as separate data 'rows', with each tab pointing back to it's parent's ID (and a lot of other information! That is what makes it hard to work with and manipulate.).

**Comment by Cal63** [comment]
Hi,

Perfect, that'll do nicely.

Thank you for your reply.

________________________________
